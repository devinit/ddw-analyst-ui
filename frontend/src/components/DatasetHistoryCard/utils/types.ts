export interface SavedQueryData {
  id: number;
  saved_query_db_table: string;
  operation: number;
  full_query: string | null;
  active: boolean;
  description: string;
  created_on: string;
  user: string;
  status: string;
}
