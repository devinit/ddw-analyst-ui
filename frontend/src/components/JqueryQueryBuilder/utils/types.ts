export type JqueryQueryBuilderFilter = {
  condition?: string;
  rules?: (JqueryQueryBuilderFilterComparator | JqueryQueryBuilderFilter)[];
};

export type JqueryQueryBuilderHaving = {
  condition?: string;
  rules?: (JqueryQueryBuilderHavingComparator | JqueryQueryBuilderHaving)[];
};

export type JqueryQueryBuilderHavingComparator = {
  id: string;
  field: string;
  type: string;
  input: string;
  operator: JqueryQueryBuilderComps;
  value?: string | number | { plain: string | number } | { column: string; aggregate: string };
  values?: { value: string; label: string }[];
};

export type JqueryQueryBuilderFilterComparator = {
  id: string;
  field: string;
  type: string;
  input: string;
  operator: JqueryQueryBuilderComps;
  value: string | number;
};
export type JqueryQueryBuilderComps =
  | 'equal'
  | 'not_equal'
  | 'less'
  | 'less_or_equal'
  | 'greater'
  | 'greater_or_equal'
  | 'between'
  | 'in';

export type JqueryQueryBuilderFieldData = {
  id: string | number;
  label: string;
  type: string;
};

export type JqueryQueryBuilderIcons = {
  add_group: string;
  add_rule: string;
  remove_group: string;
  remove_rule: string;
  error: string;
};
