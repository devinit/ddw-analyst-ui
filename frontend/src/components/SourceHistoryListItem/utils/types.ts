export interface FrozenData {
  id: number;
  parent_db_table: string;
  frozen_db_table: string | null;
  active: boolean;
  description: string;
  created_on: string;
  user: string;
  status: string;
  logs: string | null;
}
