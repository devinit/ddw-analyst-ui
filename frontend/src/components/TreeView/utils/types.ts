import { ReactElement } from 'react';

export interface TreeViewProps {
  data: Data[];
  depth?: number;

  deleteElement?: ReactElement;

  getStyleClassCb?: (node: Data) => void;

  isCheckable?: (node: Data, depth?: number) => void;
  isDeletable?: (node: Data, depth?: number) => void;
  isExpandable?: (node: Data, depth?: number) => void;

  keywordChildren?: string;
  keywordChildrenLoading?: string;
  keywordKey?: string;
  keywordLabel?: string;

  loadingElement?: ReactElement;
  noChildrenAvailableMessage?: string;

  onCheckToggleCb?: (data: Data[], depth?: number) => void;
  onDeleteCb?: (node: Data, data: Data[], depth?: number) => void;
  onExpandToggleCb?: (node: Data, depth?: number) => void;
  onUpdateCb?: (data: unknown[], depth?: number) => void;

  transitionEnterTimeout?: number;
  transitionExitTimeout?: number;
}

export interface Data {
  id: number;
  name: string;
  isExpanded?: boolean;
  isChecked?: boolean;
  isChildrenLoading?: boolean;
  children?: Data[];
}
