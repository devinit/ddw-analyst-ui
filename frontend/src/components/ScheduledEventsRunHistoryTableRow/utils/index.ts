const status: { [char: string]: string } = {
  p: 'Pending',
  r: 'Running',
  c: 'Completed',
  e: 'Errored',
  s: 'Skipped',
};

const statusClasses: { [char: string]: string } = {
  p: 'badge-secondary',
  r: 'badge-warning',
  c: 'badge-success',
  e: 'badge-danger',
  s: 'badge-info',
};

export const convertStatus = (statusValue: string): string | undefined => {
  for (const key in status) {
    if (key === statusValue) {
      return status[key].toUpperCase();
    }
  }
};

export const getStatusClasses = (statusValue: string): string | undefined => {
  for (const key in statusClasses) {
    if (key === statusValue) {
      return statusClasses[key];
    }
  }
};
