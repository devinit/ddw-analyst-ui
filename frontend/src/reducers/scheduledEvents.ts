import { List, Map, fromJS } from 'immutable';
import { Action, Reducer } from 'redux';
import { ScheduledEvent, ScheduledEventMap } from '../types/scheduledEvents';

export interface ScheduledEventsAction extends Action {
  scheduledEvents?: ScheduledEvent[];
  activeScheduledEvent?: ScheduledEventMap;
  count: number;
  activeScheduleEventIndex?: number;
  payload: Partial<{
    limit: number;
    offset: number;
    link: string;
    search: string;
    id: number | string;
  }>;
  loading?: boolean;
}
interface State {
  loading: boolean;
  scheduledEvents: List<ScheduledEventMap>;
  activeScheduledEvent?: ScheduledEventMap;
  count: number;
  activeOperationId: number;
}
export type ScheduledEventsState = Map<keyof State, State[keyof State]>;

const prefix = 'scheduledEvents';
export const FETCH_SCHEDULED_EVENTS = `${prefix}.FETCH`;
export const FETCH_SCHEDULED_EVENTS_SUCCESSFUL = `${prefix}.FETCH_SUCCESSFUL`;
export const FETCH_SCHEDULED_EVENTS_FAILED = `${prefix}.FETCH_FAILED`;
export const SET_ACTIVE_SCHEDULED_EVENT = `${prefix}.SET_ACTIVE_SCHEDULED_EVENT`;
export const FETCH_SCHEDULED_EVENT = `${prefix}.FETCH_SCHEDULED_EVENT`;
export const FETCH_SCHEDULED_EVENT_FAILED = `${prefix}.FETCH_FAILED`;
export const DELETE_SCHEDULED_EVENT = `${prefix}.DELETE_SCHEDULED_EVENT`;
export const DELETE_SCHEDULED_EVENT_SUCCESSFUL = `${prefix}.DELETE_SCHEDULED_EVENT_SUCCESSFUL`;
export const DELETE_SCHEDULED_EVENT_FAILED = `${prefix}.DELETE_SCHEDULED_EVENT_FAILED`;

export const defaultState: ScheduledEventsState = fromJS({
  loading: false,
  scheduledEvents: [],
  activeScheduledEventId: 1,
  limit: 10,
  offset: 0,
});

export const scheduledEventsReducer: Reducer<ScheduledEventsState, ScheduledEventsAction> = (
  state = defaultState,
  action,
) => {
  if (action.type === FETCH_SCHEDULED_EVENTS || action.type === FETCH_SCHEDULED_EVENT) {
    return state.set('loading', true);
  }
  if (action.type === FETCH_SCHEDULED_EVENTS_SUCCESSFUL && action.scheduledEvents) {
    return state.withMutations((map) =>
      map
        .set('loading', false)
        .set('scheduledEvents', fromJS(action.scheduledEvents))
        .set('count', action.count),
    );
  }
  if (
    action.type === FETCH_SCHEDULED_EVENTS_FAILED ||
    action.type === FETCH_SCHEDULED_EVENT_FAILED
  ) {
    return state.set('loading', false);
  }
  if (action.type === SET_ACTIVE_SCHEDULED_EVENT) {
    return state.set('activeScheduledEvent', action.activeScheduledEvent).set('loading', false);
  }

  return state;
};
