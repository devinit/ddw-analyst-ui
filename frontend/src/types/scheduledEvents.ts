import { Map } from 'immutable';

export interface ScheduledEvent {
  id: number;
  name: string;
  description: string | null;
  enabled: boolean;
  interval: number;
  interval_type: string;
  repeat: string;
}

export type ScheduledEventMap = Map<keyof ScheduledEvent, ScheduledEvent[keyof ScheduledEvent]>;
