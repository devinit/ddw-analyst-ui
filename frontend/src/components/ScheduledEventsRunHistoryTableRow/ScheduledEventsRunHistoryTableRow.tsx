import classNames from 'classnames';
import moment from 'moment';
import React, { FunctionComponent } from 'react';
import { Button } from 'react-bootstrap';
import { ScheduledEventRunHistory } from '../../types/scheduledEvents';
import { convertStatus, getStatusClasses } from './utils';

interface ScheduledEventsRunHistoryTableRowProps {
  history: ScheduledEventRunHistory;
  onViewLogs?: (info: string) => void;
}

export const ScheduledEventsRunHistoryTableRow: FunctionComponent<
  ScheduledEventsRunHistoryTableRowProps
> = ({ history, onViewLogs }) => {
  const badgeClass = getStatusClasses(history.status);
  const onClickInfo = () => {
    if (onViewLogs && history.logs) {
      onViewLogs(history.logs);
    }
  };

  return (
    <tr>
      <td>
        <span className={classNames('badge', badgeClass)}>{convertStatus(history.status)}</span>
      </td>
      <td>{moment.utc(history.start_at).format('LLL')}</td>
      <td data-testid="history-end-date">
        {history.ended_at ? moment.utc(history.ended_at).format('LLL') : ''}
      </td>
      <td>
        {history.logs && onViewLogs ? (
          <Button
            size="sm"
            variant="link"
            className="btn-danger"
            onClick={onClickInfo}
            data-testid="logs-button"
          >
            <i className="material-icons" data-testid="info-trigger">
              info
            </i>{' '}
            <span>Logs</span>
          </Button>
        ) : null}
      </td>
    </tr>
  );
};
