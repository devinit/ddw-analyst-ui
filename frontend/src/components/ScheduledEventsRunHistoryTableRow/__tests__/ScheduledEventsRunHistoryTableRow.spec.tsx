import React from 'react';
import renderer from 'react-test-renderer';
import { ScheduledEventsRunHistoryTableRow } from '../ScheduledEventsRunHistoryTableRow';
import { cleanup } from '@testing-library/react';
import { ScheduledEventRunHistory } from '../../../types/scheduledEvents';

afterEach(cleanup);

const history: ScheduledEventRunHistory = {
  scheduled_event: 1,
  start_at: '',
  status: 'pending',
};

it('should take a snapshot', () => {
  const tree = renderer.create(<ScheduledEventsRunHistoryTableRow history={history} />).toJSON();
  expect(tree).toMatchSnapshot();
});
