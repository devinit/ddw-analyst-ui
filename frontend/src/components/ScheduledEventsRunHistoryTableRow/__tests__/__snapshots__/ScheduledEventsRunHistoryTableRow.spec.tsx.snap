// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<tr>
  <td>
    <span
      className="badge"
    />
  </td>
  <td>
    Invalid date
  </td>
  <td>
    
  </td>
  <td />
</tr>
`;
